### DISCUSSION TITLE
Uruchamianie testów zaległych zadań

### DISCUSSION THREAD

**Discussion by PixelSculptor** [przeprogramowani/advent-of-frontend#28] [Pierwsze kroki] [answered]
Hej, w jaki sposób uruchomić testy z poprzednich dni? Nadrabiam zadanie z wczoraj i nie mogę poprawnie uruchomić testów dla zadania, ponieważ szuka testów do zadania z obecnego dnia  (jeśli dobrze rozumiem) i dostaje taki komunikat w terminalu:
```
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/kacper/Documents/Projekty/advent-of-frontend
  24 files checked.
  testMatch: **/__tests__/**/*.[jt]s?(x), **/?(*.)+(spec|test).[tj]s?(x) - 9 matches
  testPathIgnorePatterns: /node_modules/ - 24 matches
  testRegex:  - 0 matches
Pattern: ./tasks/2023-12-09/index.test.ts - 0 matches

```

**Comment by przeprogramowani** [comment, marked as answer]
Obecnie masz dwie opcje:
```
    "test": "jest ./tasks/$(date +'%Y-%m-%d')/index.test.ts",
    "test:all": "jest"
```

Skryptami możesz uruchomić albo wszystkie, albo dzisiejsze - niestety nie ma na dzisiaj wsparcia dla wybranego dnia wstecz.

**Reply by kewinzaq1** [reply]
@przeprogramowani @PixelSculptor Jest też trzecia opcja :). Możesz zainstalować dodatek `Jest` do vscode / narzędzia od Jetbrains mają wbudowane i kliknąć przy teście `run` lub z palety komend uruchomić test.

**Comment by dominikbulaj** [comment]
@PixelSculptor 
Ja sobie dopisałem NPM Script, w którym podmieniam datę i uruchamiam tylko jeden "dzienny" zestaw testów:
```
"test:day": "jest ./tasks/2023-12-09/index.test.ts",
```

![CleanShot 2023-12-11 at 08 40 36@2x](https://github.com/przeprogramowani/advent-of-frontend/assets/1233407/c8f3f5ea-346a-4d46-b598-354cb6a26769)
